CMakeLists.txt

```diff
@@ -164,7 +164,7 @@ if (NOT PG_CONFIG)
   message(FATAL_ERROR "Unable to find 'pg_config'")
 endif ()
 
-find_package(Git)
+find_package(Git REQUIRED)
```

Comment:
Unclear if this is a great idea for windows builds. Then we are adding Git as a dependency for building on all platforms.

Reply:
Since this function is about providing git information, we already have that (it passes the Windows build in the CI). An option would be to make Git optional but then require building from a source tar, which would include the generated file with git information, or Git, and not make the function available otherwise.

Reply:
I am not sure I am following. AFAIK, Git was not required previously---at least that was not the intention. That things pass _our_ CI is a different matter since we control that environment and can install whatever dependencies we like. But I am not sure we should require all those dependencies for other source builds. Compiling from a tar ball is one example.

Reply:
Right now, the function is defined to return something else than the git version if git does not exists, which I think is wrong since we use this for debugging we do not get usable information if the functions invent a commit tag that it is built from (the project version).

I added the requirements since it seemed the intention was to always provide that function, but a better option would be to remove the function if git is not installed and not require it Git when building from a non-git source.

When building from a source tarball, configuration files are usually included and since the git information is in a configuration file, that would be included in the source tarball. However, it seems we have not configured for building source tarballs, which would explain that the function uses the project version instead of the git tag when git is not installed.

I would suggest the following then (for this PR):

- Do not require Git.
- If Git is not installed, do not provide `get_git_commit`. This will make it clear that the extension is built is from a non-git source build. The extension version is still available in `\dx` so we can get that information regardless.
- If Git is installed, provide the `get_git_commit`. This will make it clear that it is built from a git build and what git commit that the extension is built from, regardless of whether it is a Docker image, a package, or a source build in a git repository.

Reply:
My only concern is that we should support tar-ball builds. Even if Git is installed when you build tar-balls, Git won't provide useful information since you are not in a repo. Thus, requiring GIt seems wrong. We should fall back to other information sources, like reading the version from `version.config` (perhaps with a useful suffix) or similar.

I think it is fine to do best effort w.r.t. version info, but for "released" tar balls we should be able to include a pre-generated version file. Note, however, that, e.g., GitHub provides tar-ball downloads automatically by simply archiving the Git repo: https://github.com/timescale/timescaledb/archive/2.0.0-rc3.tar.gz. Such a tar ball won't include a version.h file and in that case I think we must fall back to`version.config` as outlined above.

I am not sure what you mean by "not provide `get_git_commit`"--- I think this function will exist regardless, although the returned info can of course vary.

Reply:
Changed the PR to not require Git and not define the function if the git information cannot be retrieved.